src/mds/CDir.cc

```diff
@@ -608,10 +608,8 @@ void CDir::unlink_inode_work( CDentry *dn )
       dn->unlink_remote(dn->get_linkage());
 
     dn->get_linkage()->set_remote(0, 0);
-  } else {
+  } else if (dn->get_linkage()->is_primary()) {
```

Comment:
null dentry Shouldn't be here，so add assert(!dn->get_linkage()->is_null()) in the else branch